app/src/features/apiClient/screens/apiClient/components/modals/postmanExportModal/PostmanExportModal.tsx

```
@@ -0,0 +1,169 @@
+  schema_version: string;
+  records: ExportRecord[];
+}
+
+const COLLECTIONS_SCHEMA_VERSION = "1.0.0";
+
+export const PostmanExportModal: React.FC<PostmanExportModalProps> = ({ recordsToBeExported, isOpen, onClose }) => {
+  const [isApiRecordsProcessed, setIsApiRecordsProcessed] = useState(false);
+  const [exportData, setExportData] = useState<RequestlyExportData>({
+    schema_version: COLLECTIONS_SCHEMA_VERSION,
+    records: [],
+  });
+  const [fileInfo, setFileInfo] = useState<{ label: string; type: string }>({ label: "", type: "" });
+
+  const handleExport = useCallback(async () => {
+    try {
+      // Convert Requestly collection to Postman format
+      const result = postmanCollectionExporter(exportData);
+
+      // Check if result is a Promise (multiple collections case)
+      if (result instanceof Promise) {
+        // Multiple collections - will return ZIP
+        const multipleResult = await result;
+        if (multipleResult.type === "multiple") {
+          // Create a blob from the zip data and download it
+          const blob = new Blob([multipleResult.zipData], { type: "application/zip" });
+          const fileName = `Postman-${fileInfo.label}-export-${getFormattedDate("DD_MM_YYYY")}.zip`;
+
+          // Create a download link
+          const url = URL.createObjectURL(blob);
+          const link = document.createElement("a");
+          link.href = url;
+          link.download = fileName;
+          document.body.appendChild(link);
+          link.click();
+          document.body.removeChild(link);
+          URL.revokeObjectURL(url);
+        }
+      } else {
+        // Single collection - return JSON
+        if (result.type === "single") {
+          const fileContent = JSON.stringify(result.collection, null, 2);
+          const fileName = `Postman-${fileInfo.label}-export-${getFormattedDate("DD_MM_YYYY")}.json`;
+          fileDownload(fileContent, fileName, "application/json");
+        }
+      }
+
```

If `result` is a Promise but doesn't resolve to an object with type: `multiple`, the code will silently fail without handling the error case.

---

Also is it really necessary to check for Promise instance? Can't we just await for the `result` and check for `result.type`